wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -274,7 +259,8 @@ public final class ZMUserSession: NSObject {
             onNewCRLsDistributionPointsSubject: onNewCRLsDistributionPointsSubject
         )
 
-        cRLsDistributionPointsObserver.startObservingNewCRLsDistributionPoints(
+        // TODO: why is this observer started two times? first time is 'setupCertificateRevocationLists'!
+        cRLsDistributionPointsObserver?.startObservingNewCRLsDistributionPoints(
             from: onNewCRLsDistributionPointsSubject.eraseToAnyPublisher()
         )
```

Comment:
@KaterinaWire @David-Henner can you help me with this question?

Reply:
we're adding a different publisher. in `setupCertificateRevocationLists` the publisher is the publisher from `mlsService`. Here it's another publisher, `onNewCRLsDistributionPointsSubject` that is passed to `E2EIKeyPackageRotator` and `E2EIRepository`.

Technically we could have only one publisher and pass it down to all the dependencies

Reply:
Okay sounds good so far, I think we need someway to make sure that `setupCertificateRevocationLists()` and this are executed in the right order. Maybe we can have a short pairing session?

Reply:
@David-Henner since I am not really able to resolve the timing issue now, I've added just an assertion in 0624c4fd98dab07bad0e162c4a6da5efe9c30beb. This whole construction with lazy vars, that are dependent one each other is quite fragile.
Probably we can make a follow up PR here to clean this up, what do you think? cc @KaterinaWire

Reply:
Agree